src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java

```diff
@@ -102,4 +107,105 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, Ex
             }
         });
     }
+
+    private static class UpdateListMethodInvocations extends JavaIsoVisitor<ExecutionContext> {
+        private static final String ITERABLE_FQN = "java.lang.Iterable";
+        private static final String GET_ITEMS_METHOD = "getItems";
+        private final String parameterName;
+
+        public UpdateListMethodInvocations(String parameterName) {
+            this.parameterName = parameterName;
+        }
+
+        private static final MethodMatcher LIST_MATCHER = new MethodMatcher("java.util.List *(..)", true);
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            J.MethodInvocation mi = super.visitMethodInvocation(method, executionContext);
+            if (LIST_MATCHER.matches(mi) && mi.getSelect() instanceof J.Identifier) {
+                if (isParameter((J.Identifier) mi.getSelect())) {
+                    assert mi.getPadding().getSelect() != null;
+                    // No need to take care of typing here, since it's going to be printed and parsed on the JavaTemplate later on.
+                    mi = mi.withSelect(newGetItemsMethodInvocation(mi.getPadding().getSelect()));
+                }
+            }
+            if (mi.getMethodType() != null && !ITEM_WRITER_MATCHER.matches(mi)) {
```

Comment:
Matcher already does that null check.
```suggestion
            if (!ITEM_WRITER_MATCHER.matches(mi)) {
```